lib/pbench/server/cache_manager.py

```diff
@@ -491,12 +574,13 @@ def _get_metadata(tarball_path: Path) -> Optional[JSONOBJECT]:
         """
         name = Dataset.stem(tarball_path)
         try:
-            data = Tarball.extract(tarball_path, f"{name}/metadata.log")
-        except BadDirpath:
+            data: IO[bytes] = Tarball.extract(tarball_path, f"{name}/metadata.log")
+        except CacheExtractBadPath:
             return None
         else:
             metadata_log = MetadataLog()
             metadata_log.read_file(e.decode() for e in data)
+            data.close()
             metadata = {s: dict(metadata_log.items(s)) for s in metadata_log.sections()}
             return metadata
```

Comment:
There's no unit test coverage for this function.

Reply:
I find it too awkward to try to get coverage data before pushing a PR, and often don't bother. But that's a surprise since this function is by no means new.

Reply:
> I find it too awkward to try to get coverage data before pushing a PR

Agreed.  And, in fact, it's awkward to get until all the tests are passing!  🙁  But, I didn't know whether you would go look.

> that's a surprise since this function is by no means new.

Agreed.  And, I'm pretty sure we had coverage for this before, so I was thinking that it must be some artifact of you change.  😕